File: Week_01/G20200343030410/move-zeroes.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        for j in range(1, len(nums)):
            if nums[i] == 0:
                if nums[j] != 0:
                    nums[i], nums[j] = nums[j], nums[i]
                else:
                    continue
            i += 1
# @lc code=end

File: Week_01/G20200343030628/MoveZeroes.java
public class MoveZeroes {
    public void moveZeroes(int [] nums) {
        //排除两种特殊情况
        if (nums == null || nums.length == 1) {
            return;
        }
        int count = 0;
        for (int i = 0; i < nums.length; i ++) {
            if (nums[i] == 0) {
                count++;
            } else if (count >0) {
                //交换位置
                nums[i-count] = nums[i];
                nums[i]  =  0;
            }
        }
    }

}
